Frameworks/CoreGraphics/CGColorSpace.mm

```diff
@@ -38,135 +39,148 @@
 const CFStringRef kCGColorSpaceITUR_2020 = static_cast<CFStringRef>(@"kCGColorSpaceITUR_2020");
 const CFStringRef kCGColorSpaceROMMRGB = static_cast<CFStringRef>(@"kCGColorSpaceROMMRGB");
 
-static IWLazyClassLookup _LazyUIColor2("UIColor");
+static woc::StrongCF<CGColorSpaceRef> s_rgbColorSpace;
+static woc::StrongCF<CGColorSpaceRef> s_grayColorSpace;
 
-@interface CGNSColorSpace : _CGLifetimeBridgingType
-@end
+struct __CGColorSpace : CoreFoundation::CppBase<__CGColorSpace> {
+    __CGColorSpace(CGColorSpaceModel model, CGColorSpaceRef baseSpace = nullptr, size_t lastIndex = 0)
+        : _colorSpaceModel(model), _baseColorSpace(baseSpace), _lastColorIndex(lastIndex) {
+    }
 
-@implementation CGNSColorSpace
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wobjc-missing-super-calls"
-- (void)dealloc {
-    delete (__CGColorSpace*)self;
-}
-#pragma clang diagnostic pop
-@end
+    __CGColorSpace(CGColorSpaceModel model, CGColorSpaceRef baseSpace, size_t lastIndex, const unsigned char* colorTable)
+        : __CGColorSpace(model, baseSpace, lastIndex) {
+        size_t size = CGColorSpaceGetNumberOfComponents(baseSpace) * (lastIndex + 1);
+        _palette.reset(new char[size]);
+        memcpy(_palette.get(), colorTable, size);
```

Comment:
What benefits does using a `unique_ptr<char>` get us over a vector? The vector can be copy-initialized without memcpy, and that would let you delete the entire constructor body.

When in doubt, try to construct things in the initializer list :smile: